src/AccountingModule.sol

```diff
@@ -0,0 +1,448 @@
+            / (currentTimestamp - previousTimestamp);
+    }
+
+    /// LOSS ///
+
+    /**
+     * @notice Process losses by burning accounting tokens
+     * @param amount losses to burn
+     */
+    function processLosses(uint256 amount) external onlyRole(LOSS_PROCESSOR_ROLE) checkAndResetCooldown {
+        AccountingModuleStorage storage s = _getAccountingModuleStorage();
+        uint256 totalSupply = s.accountingToken.totalSupply();
+        if (totalSupply < 10 ** s.accountingToken.decimals()) revert TvlTooLow();
+
+        // check bound on losses
+        if (amount > totalSupply * s.lowerBound / DIVISOR) {
+            revert LossLimitsExceeded(amount, totalSupply * s.lowerBound / DIVISOR);
+        }
+
+        s.accountingToken.burnFrom(STRATEGY, amount);
+        IVault(STRATEGY).processAccounting();
+
+        createStrategySnapshot();
+    }
+
+    /// ADMIN ///
+
+    /**
+     * @notice Set target APY to determine upper bound. e.g. 1000 = 10% APY
+     * @param targetApyInBips in bips
+     * @dev hard max of 100% targetApy
+     */
+    function setTargetApy(uint256 targetApyInBips) external onlyRole(SAFE_MANAGER_ROLE) {
+        AccountingModuleStorage storage s = _getAccountingModuleStorage();
+        if (targetApyInBips > DIVISOR) revert InvariantViolation();
+
+        emit TargetApyUpdated(targetApyInBips, s.targetApy);
+        s.targetApy = targetApyInBips;
+    }
+
+    /**
+     * @notice Set lower bound as a function of tvl for losses. e.g. 1000 = 10% of tvl
+     * @param _lowerBound in bips, as a function of % of tvl
+     * @dev hard max of 50% of tvl
+     */
+    function setLowerBound(uint256 _lowerBound) external onlyRole(SAFE_MANAGER_ROLE) {
+        AccountingModuleStorage storage s = _getAccountingModuleStorage();
+        if (_lowerBound > (MAX_LOWER_BOUND)) revert InvariantViolation();
+
+        emit LowerBoundUpdated(_lowerBound, s.lowerBound);
+        s.lowerBound = _lowerBound;
+    }
+
+    /**
+     * @notice Set cooldown in seconds between every processing of rewards/losses
+     * @param cooldownSeconds_ new cooldown seconds
+     */
+    function setCooldownSeconds(uint16 cooldownSeconds_) external onlyRole(SAFE_MANAGER_ROLE) {
+        AccountingModuleStorage storage s = _getAccountingModuleStorage();
+        emit CooldownSecondsUpdated(cooldownSeconds_, s.cooldownSeconds);
+        s.cooldownSeconds = cooldownSeconds_;
+    }
+
+    /**
+     * @notice Set a new safe address
+     * @param newSafe new safe address
+     */
+    function setSafeAddress(address newSafe) external virtual onlyRole(SAFE_MANAGER_ROLE) {
+        AccountingModuleStorage storage s = _getAccountingModuleStorage();
+        if (newSafe == address(0)) revert ZeroAddress();
+        emit SafeUpdated(newSafe, s.safe);
+        s.safe = newSafe;
+    }
+
+    /// VIEWS ///
+
+    function accountingToken() external view returns (IAccountingToken) {
```

Comment:
replace this function body with:

```solidity
return _getAccountingModuleStorage().accountingToken
```

Also applies to all the other functions below this one.